### PR TITLE
[WIP] Run clang-format on src folder

### PR DESCRIPTION
**Pull request by frizensami**
Command: `find src/ -iname *.h -o -iname *.cpp | xargs clang-format -i --style=Google`

Does this look as expected?

### CODE REVIEW

**Comment by frizensami** [comment]
Apparently it was due to `vector` being included in my .cpp file instead of my .h file somewhere. Not sure why clang caught that only after the formatting changes.

**Comment by jethrokuan** [comment]
I'd prefer if you included it in both, since both actually use it.

**Comment by jethrokuan** [comment]
There's quite a bit of changes to my code, which I had preformatted with clang-format as well :thinking:

**Comment by frizensami** [comment]
> I'd prefer if you included it in both, since both actually use it.

My .cpp file includes the corresponding .h file which includes `vector` already - is there an advantage to including it again in the .cpp file?

**Comment by jethrokuan** [comment]
> My .cpp file includes the corresponding .h file which includes vector already - is there an advantage to including it again in the .cpp file?

No, I don't know what the convention is tbh.

> Do you get any changes if you run your clang-format with --style=Google instead?

Not as extensive. Shall give an example

File: src/spa/src/simple_parser/parser.h [hunk 1]
```diff
@@ -12,17 +12,17 @@ namespace Simple {
...
 
 // NumberExprAST - Expression class for numeric literals
 class NumberExpr : public Expr {
  public:
   std::string Val;
-  NumberExpr(const std::string& Val) : Val(Val){};
-  bool operator==(const Expr& other) const {
-    auto casted_other = dynamic_cast<const NumberExpr*>(&other);
+  NumberExpr(const std::string &Val) : Val(Val){};
+  bool operator==(const Expr &other) const {
```
Comment (jethrokuan) [inline src/spa/src/simple_parser/parser.h]: These shifting of pointer thingies to the start of the variable doesn't happen for me, I think the new Google style prefers it the other way around.

**Comment by frizensami** [comment]
OK, am installing clang-5.0 and clang-format-5.0, will let you know once I've tried it.

**Comment by frizensami** [comment]
Could not compile with `catch.hpp`  + `clang-5.0` (clang v 5.0.0.*). Trying clang-6.0.

**Comment by frizensami** [comment]
Similar for clang-6.0. Errors are as such:

```
In file included from /home/sriram/Projects/cs3203/code/src/integration_testing/src/main.cpp:2:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:631:53: error: no template named 'bool_constant' in namespace 'std'; did you mean '__bool_constant'?
        inline constexpr auto is_unique<T, Rest...> = std::bool_constant<
                                                      ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:93:5: note: '__bool_constant' declared here
    using __bool_constant = integral_constant<bool, __v>;
    ^
In file included from /home/sriram/Projects/cs3203/code/src/integration_testing/src/main.cpp:2:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:10: error: no template named 'is_same_v' in namespace 'std'; did you mean 'is_same'?
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:1450:12: note: 'is_same' declared here
    struct is_same
           ^
In file included from /home/sriram/Projects/cs3203/code/src/integration_testing/src/main.cpp:2:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:29: error: expected '(' for function-style cast or type construction
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~~~~~~~~~~~~~~~~~~~ ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestQueryEvaluator.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:631:53: error: no template named 'bool_constant' in namespace 'std'; did you mean '__bool_constant'?
        inline constexpr auto is_unique<T, Rest...> = std::bool_constant<
                                                      ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:93:5: note: '__bool_constant' declared here
    using __bool_constant = integral_constant<bool, __v>;
    ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestQueryEvaluator.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:10: error: no template named 'is_same_v' in namespace 'std'; did you mean 'is_same'?
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:1450:12: note: 'is_same' declared here
    struct is_same
           ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestQueryEvaluator.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:29: error: expected '(' for function-style cast or type construction
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~~~~~~~~~~~~~~~~~~~ ^
3 warnings generated.
[ 61%] Linking CXX executable autotester
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestDeclaration.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:631:53: error: no template named 'bool_constant' in namespace 'std'; did you mean '__bool_constant'?
        inline constexpr auto is_unique<T, Rest...> = std::bool_constant<
                                                      ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:93:5: note: '__bool_constant' declared here
    using __bool_constant = integral_constant<bool, __v>;
    ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestDeclaration.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:10: error: no template named 'is_same_v' in namespace 'std'; did you mean 'is_same'?
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:1450:12: note: 'is_same' declared here
    struct is_same
           ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestDeclaration.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:29: error: expected '(' for function-style cast or type construction
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~~~~~~~~~~~~~~~~~~~ ^
[ 61%] Built target autotester
[ 66%] Building CXX object src/unit_testing/CMakeFiles/unit_testing.dir/src/TestQueryPreProcessor.cpp.o
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestQueryPreProcessor.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:631:53: error: no template named 'bool_constant' in namespace 'std'; did you mean '__bool_constant'?
        inline constexpr auto is_unique<T, Rest...> = std::bool_constant<
                                                      ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:93:5: note: '__bool_constant' declared here
    using __bool_constant = integral_constant<bool, __v>;
    ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestQueryPreProcessor.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:10: error: no template named 'is_same_v' in namespace 'std'; did you mean 'is_same'?
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
                  ~~~~~^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/type_traits:1450:12: note: 'is_same' declared here
    struct is_same
           ^
In file included from /home/sriram/Projects/cs3203/code/src/unit_testing/src/TestQueryPreProcessor.cpp:1:
/home/sriram/Projects/cs3203/code/lib/catch.hpp:632:29: error: expected '(' for function-style cast or type construction
                (!std::is_same_v<T, Rest> && ...) && is_unique<Rest...>
```

**Older clang: clang-3.8 - has no errors during compilation. ** - just tested with it again. 

This might be just be a problem where other parts of compilation require an executable called `clang` instead of `clang-*`.

**Comment by frizensami** [comment]
Switched back to **clang 3.8**.
Installed **clang-format-6.0**

Making new PR with format using .clang-format file.

**Comment by frizensami** [comment]
Formatted with clang-format-6.0 and using the provided .clang-format file - #27

**Comment by frizensami** [comment]
Superseded by #27

**Pull request closed without merging**